server/modules/environment-state/getServiceTargetStates.js

```diff
@@ -2,93 +2,78 @@
 'use strict';
 
 let _ = require('lodash');
-let keyValueStore = require('modules/service-updater/consul/keyValueStore');
 let Enums = require('Enums');
 let co = require('co');
 let logger = require('modules/logger');
-let serviceReporter = require('modules/service-reporter');
+let targetStates = require('modules/service-targets');
+
+const HEALTH_GOOD = Enums.HEALTH_STATUS.Healthy;
+const HEALTH_BAD = Enums.HEALTH_STATUS.Error;
 
 /**
  * Generate service health info (with checks list and pass / fail)
  */
 function getServiceChecksInfo(serviceObjects) {
   // Get all health checks for all instances of this service
   var serviceChecks = _.flatMap(serviceObjects, 'HealthChecks');
-   
   var checksGrouped = _.groupBy(serviceChecks, 'Name');
   return _.map(checksGrouped, function (checks, checkName) {
-     
     // If some instances checks failed for a given check, mark as failed
     // also, don't count in instance into working
-    if (_.some(checks, { Status: 'critical' })) {
-      return {
-        Name: checks[0].Name,
-        Status: Enums.HEALTH_STATUS.Error,
-      };
-    } else {
-      return {
-        Name: checks[0].Name,
-        Status: Enums.HEALTH_STATUS.Healthy,
-      };
-    }
-  });
-}
-
-function getServiceOverallHealth(healthChecks, instances) {
-  if (_.some(healthChecks, { Status: Enums.HEALTH_STATUS.Error })) {
     return {
-      Status: Enums.HEALTH_STATUS.Error,
+      Name: checks[0].Name,
+      Status: _.some(checks, { Status: 'critical' }) ? HEALTH_GOOD : HEALTH_BAD
     };
-  } else {
-    return {
-      Status: Enums.HEALTH_STATUS.Healthy
-    };
-  }
+  });
 }
 
-function* getServicesState(environmentName, runtimeServerRoleName, instances) {
-  let servicesList = yield serviceReporter.getServicesList(environmentName, runtimeServerRoleName);
+function getServiceOverallHealth(healthChecks) {
+  return {
+    Status:_.some(healthChecks, { Status: HEALTH_BAD }) ? HEALTH_BAD : HEALTH_GOOD
```

Comment:
👍 , just formatting. We'll have also third state, WARNING, in which case logic will be more complex